### PR TITLE
Stanza Receiver Implementation

### PR DESCRIPTION
**Pull request by djaglowski**
**Description:**
This PR is a continuation of #1096  and #1117 (also see #1092 for additional context), and adds a concrete implementation for the receiver.

Left out, is the actual conversion from stanza's log format to pdata format.

**Testing:**
Receiver is fully unit tested.
Includes a `mocks_test.go` file that helps test some edge cases in receiver startup.

### CODE REVIEW

File: exporter/honeycombexporter/honeycomb_test.go
Comment (djaglowski) [inline exporter/honeycombexporter/honeycomb_test.go:291]: This should be removed before merging this PR. (Looks like #1221 may address the problem seen here.)

**Comment by djaglowski** [comment]
@keitwb I've backported all the relevant changes from #1125. Do you mind giving this a re-review? Thanks.

**Comment by djaglowski** [comment]
Hi @keitwb. I saw that you reapproved #1125, but I believe we need to process this PR first. Mind approving this one? Thanks again.

**Comment by djaglowski** [comment]
@keitwb Sorry to keep hassling you, but this PR is the one that needs approval. Thanks again!

File: exporter/honeycombexporter/honeycomb_test.go
Comment (keitwb) [reply exporter/honeycombexporter/honeycomb_test.go:291]: Does this still need to be removed?

**Review by keitwb — APPROVED** [review]
Looks good, just be sure the test isn't skipped once everything is merged across all the PRs.

File: exporter/signalfxexporter/translation/translator_test.go
Comment (djaglowski) [inline exporter/signalfxexporter/translation/translator_test.go:1690]: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/1232

File: exporter/honeycombexporter/honeycomb_test.go
Comment (djaglowski) [reply exporter/honeycombexporter/honeycomb_test.go]: This issue was resolved, but there appears to be another. Commented below.

**Comment by djaglowski** [comment]
@tigrannajaryan, I believe this is ready for your review now. Thank you in advance.

File: exporter/signalfxexporter/translation/translator_test.go
Comment (djaglowski) [reply exporter/signalfxexporter/translation/translator_test.go]: This has been resolved.

**Comment by djaglowski** [comment]
How can we move this forward? This PR has been open for 2+ weeks and has only received trivial feedback.

**Comment by keitwb** [comment]
@djaglowski Looks fine to me, but I don't have power to merge.  @tigrannajaryan

**Review by tigrannajaryan — APPROVED** [review]

**Comment by tigrannajaryan** [comment]
Thanks @djaglowski and sorry for the delay, I was away for a few days.

**Pull request merged**
